### ISSUE TITLE
missing includes when compiling as library with cmake

### ISSUE THREAD

**Issue by gpenazzi** [dftbplus/mpifx#14]
I am compiling as shared lib and it seems that the MPI includes are missing. Am I doing something wrong? Fixed locally by adding `target_include_directories(mpifx PRIVATE ${MPI_Fortran_MODULE_DIR})` 

Related, wouldn't it make sense to also link the MPI libs as `ttarget_link_libraries(mpifx PRIVATE ${MPI_Fortran_LIBRARIES})` when building as shared lib?

**Comment by bhourahine** [comment]
Unfortunately the main branch library is serial only at the moment as the DFTB+ API doesn't have routines to handle the passing of MPI communicators. See #326 for the in-progress extension to cover this.

**Comment by gpenazzi** [comment]
I'm not sure I get it. My problem is that `cmake -DBUILD_SHARED_LIBS=TRUE` will not produce a vaiid Makefile because of the missing includes. Is stand-alone compilation via cmake not supported yet, or not planned either? Would it be an option to add an option to enable stand-alone compilation?

**Comment by bhourahine** [comment]
What functionality are you requesting in config.cmake? At the moment, libNEGF doesn't work with shared library compilation and currently has to be serial if you need a library built. Using the default config.cmake, what happens if you try:
cmake -DBUILD_SHARED_LIBS=TRUE -DCMAKE_TOOLCHAIN_FILE=../sys/gnu.cmake ..

**Comment by gpenazzi** [comment]
>At the moment, libNEGF doesn't work with shared library compilation and currently has to be serial if you need a library built. Using the default config.cmake, what happens if you try:
> cmake -DBUILD_SHARED_LIBS=TRUE -DCMAKE_TOOLCHAIN_FILE=../sys/gnu.cmake ..

Maybe there's a misunderstanding. I'm not talking about the DFTB+ build chain. I would like to be able to compile mpifx alone as shared lib using cmake. 
 
libNEGF does compile stand-alone as shared lib (in the current master), and it is loaded as such in a python wrapper I'm working on (https://github.com/gpenazzi/pynegf). However when used with MPI it has mpifx as dependency (I have a related open PR in libnegf). I'd like to compile it separately and link it. If this is not intended usage of mpifx I can work around it, but I thought that it might a desirable feature.

I could compile it with this diff: [mpifx_shared.txt](https://github.com/dftbplus/mpifx/files/4014447/mpifx_shared.txt)

I hope I explained myself better now. Does it sound like something you'd like to have? I tested 
also for a DFTB+ compilation with the gnu toolchain (and MPI on) and it seems ok.
